internal/sourcemap/sourcemap.go

```diff
@@ -639,6 +648,22 @@ func (b *ChunkBuilder) GetLineAndColumnForLocation(location loc.Loc) []int {
 	return []int{originalLine + 1, originalColumn + 1}
 }
 
+func (b *ChunkBuilder) OffsetAt(location loc.Loc) int {
+	lineAndColumn := b.GetLineAndColumnForLocation(location)
+	line := lineAndColumn[0] - 1
+	column := lineAndColumn[1] - 1
+
+	// Collect the length of every line before this one
+	offset := 0
+	for i := 0; i < line; i++ {
+		currentLine := b.lineOffsetTables[i]
+		offset += currentLine.utf16LineLength
+	}
```

Comment:
Perhaps if we keep `utf16LineLength` as the cumulative line length from the previous lines would help perf here a bit? That way we only need to access the the utf16 length once instead of calculating it everytime `OffsetAt` is called.

Reply:
`utf16LineLength` is stored in the line offset table, it doesn't get calculated when `OffsetAt` is called. It ends up being a pretty simple int + int operation